crates/provider/src/shared/types.rs

```diff
@@ -0,0 +1,185 @@
+use std::{
+    collections::HashMap, fs::File, io::Write, os::unix::process::ExitStatusExt, path::PathBuf,
+    process::ExitStatus, vec,
+};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum ZombieRole {
+    Temp,
+    Node,
+    BootNode,
+    Collator,
+    CumulusCollator,
+    Authority,
+    FullNode,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum PortName {
+    Prometheus,
+    Rpc,
+    RpcWs,
+    P2P,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum ImagePullPolicy {
+    IfNotPresent,
+    Never,
+    Always,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct FileMap {
+    local_file_path:  PathBuf,
+    remote_file_path: PathBuf,
+    unique:           bool,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct RunCommandResponse {
+    pub exit_code: ExitStatus,
+    pub std_out:   Vec<u8>,
+    pub std_err:   Option<Vec<u8>>,
+}
+
+impl RunCommandResponse {
+    pub fn default() -> Self {
+        Self {
+            exit_code: ExitStatus::from_raw(0),
+            std_out:   vec![],
+            std_err:   None,
+        }
+    }
+}
+
+#[derive(Debug, Default, Clone, PartialEq)]
+pub struct NativeRunCommandOptions {
+    pub allow_fail: bool,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct NamespaceLabels {
+    job_id:       String,
+    project_name: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct NamespaceMetadata {
+    pub name:   String,
+    pub labels: Option<NamespaceLabels>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct NamespaceDef {
+    pub api_version: String,
+    pub kind:        String,
+    pub metadata:    NamespaceMetadata,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct PodLabels {
+    pub zombie_role: ZombieRole,
+    pub app:         String,
+    pub zombie_ns:   String,
+    pub name:        String,
+    pub instance:    String,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct PodMetadata {
+    pub name:      String,
+    pub namespace: String,
+    pub labels:    PodLabels,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct PodSpec {
+    pub cfg_path:  String,
+    pub data_path: String,
+    pub ports:     Vec<Port>,
+    pub command:   String,
+    pub env:       ProcessEnvironment,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct PodDef {
+    pub metadata: PodMetadata,
+    pub spec:     PodSpec,
+}
+
+type ProcessEnvironment = HashMap<String, String>;
```

Comment:
I don't know if it's possible to use ```HashMap::from([("foo", "bar"), ("bar"baz")]```, with this type ?

If not, it will force us to first create a hash map and insert k/v individually, which is not practical, maybe we could replace this with:

```rust
struct EnvVar {
   name: String,
   value: String
}

impl From<(&str, &str)> for EnvVar {
    fn into(value: (&str, &str) -> Self {
        Self { name: value.0.into(), value: value.1.into() }    
    }
}
```

this way we can use this directly with ```let environment: Vec<EnvVar> = vec![("foo", "bar").into(), ("bar", "baz).into()];```